velox/connectors/hive/storage_adapters/abfs/tests/AbfsFileSystemTest.cpp

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/connectors/hive/storage_adapters/abfs/AbfsFileSystem.h"
+#include "gtest/gtest.h"
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/common/file/File.h"
+#include "velox/common/file/FileSystems.h"
+#include "velox/connectors/hive/FileHandle.h"
+#include "velox/connectors/hive/HiveConfig.h"
+#include "velox/connectors/hive/storage_adapters/abfs/AbfsReadFile.h"
+#include "velox/connectors/hive/storage_adapters/abfs/tests/AzuriteServer.h"
+#include "velox/exec/tests/utils/TempFilePath.h"
+
+#include <atomic>
+#include <random>
+
+using namespace facebook::velox;
+
+using ::facebook::velox::common::Region;
+
+constexpr int kOneMB = 1 << 20;
+static const std::string filePath = "test_file.txt";
+static const std::string fullFilePath =
+    facebook::velox::filesystems::test::AzuriteABFSEndpoint + filePath;
+
+class AbfsFileSystemTest : public testing::Test {
+ public:
+  static std::shared_ptr<const Config> hiveConfig(
+      const std::unordered_map<std::string, std::string> configOverride = {},
+      bool useAzuriteConnectionStr = true) {
+    std::unordered_map<std::string, std::string> config(
+        {{"fs.azure.account.key.test.dfs.core.windows.net", "test"},
+         {filesystems::abfs::AbfsFileSystem::kReadAbfsConnectionStr,
+          facebook::velox::filesystems::test::AzuriteConnectionString}});
+
+    if (!useAzuriteConnectionStr) {
+      config.erase(filesystems::abfs::AbfsFileSystem::kReadAbfsConnectionStr);
+    }
+
+    // Update the default config map with the supplied configOverride map
+    for (const auto& item : configOverride) {
+      config[item.first] = item.second;
+      std::cout << "config " + item.first + " value " + item.second
+                << std::endl;
+    }
+
+    return std::make_shared<const core::MemConfig>(std::move(config));
+  }
+
+ public:
+  static void SetUpTestSuite() {
```

Comment:
Do you know if we can run multiple instances of the AzuriteServer?
If yes, then it is better to create a server for each test via ` void SetUp() override`
We need to make this change for S3 as well.
See the discussion here: https://github.com/facebookincubator/velox/pull/6309#discussion_r1329363295

Reply:
done to launch azurite server instance per each test